nullaway/src/main/java/com/uber/nullaway/generics/GenericsChecks.java

```diff
@@ -102,6 +104,22 @@ private static final class InferenceFailure implements MethodInferenceResult {
   private final Config config;
   private final Handler handler;
 
+  /**
+   * Was this code invoked from dataflow? Used to avoid infinite recursion.
+   *
+   * @see
+   *     com.uber.nullaway.dataflow.AccessPathNullnessPropagation#genericReturnIsNullable(MethodInvocationNode)
+   */
+  private boolean calledFromDataflow = false;
```

Comment:
Maybe not quite yet, but I wonder if we will end up needing a `GenericsChecks.context` object with more information specific to where `GenericsChecks` is being run on. Also, do we re-use these objects? If not, I'd consider checking if we can make this final and set it when creating the `GenericsChecks` object.